Sources/Bot/Services/MergeService.swift

```diff
@@ -0,0 +1,615 @@
+import Foundation
+import Result
+import ReactiveSwift
+import ReactiveFeedback
+
+public final class MergeService {
+    let state: Property<State>
+
+    private let logger: LoggerProtocol
+    private let github: GitHubAPIProtocol
+    private let scheduler: DateScheduler
+
+    private let pullRequestChanges: Signal<(PullRequestMetadata, PullRequest.Action), NoError>
+    private let pullRequestChangesObserver: Signal<(PullRequestMetadata, PullRequest.Action), NoError>.Observer
+
+    private let statusChecksCompletion: Signal<StatusChange, NoError>
+    private let statusChecksCompletionObserver: Signal<StatusChange, NoError>.Observer
+
+    public init(
+        integrationLabel: String,
+        statusChecksTimeout: TimeInterval = 60.minutes,
+        logger: LoggerProtocol,
+        github: GitHubAPIProtocol,
+        scheduler: DateScheduler = QueueScheduler()
+        ) {
```

Comment:
```suggestion
    ) {
```
🤓

Reply:
Overall the indentation is weird in places :) Clearly the result of a copy.

Reply:
I'll make a pass to review everything @karabatov, thanks!